internal/provider/resources/block.go

```diff
@@ -383,12 +447,22 @@ func (r *BlockResource) Update(ctx context.Context, req resource.UpdateRequest,
 		return
 	}
 
-	var data map[string]interface{}
-	resp.Diagnostics.Append(plan.Data.Unmarshal(&data)...)
+	data, diags := helpers.UnmarshalOptional(plan.Data)
+	resp.Diagnostics.Append(diags...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
+	// If the data_wo_version is different, we need to update the data payload
+	// using the content of the data_wo attribute.
+	if !plan.DataWOVersion.Equal(state.DataWOVersion) {
+		data, diags = helpers.UnmarshalOptional(config.DataWO)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
```

Comment:
im guessing this is needed because there's no way to actually tell if `data_wo` has changed by the value alone, because we dont store it

Reply:
Yep exactly, there isn't an explicit example of how to handle this in the docs but this seems to be roughly what other providers do.